devito/arguments.py

```diff
@@ -209,8 +206,23 @@ def dtype(self):
 
     @cached_property
     def rtargs(self):
-        size = ScalarArgument("%s_size" % self.name, self, max)
-        return [size]
+        size = ScalarArgument(self.size_name, self, max)
+        start = ScalarArgument(self.start_name, self, max, 0)
+        end = ScalarArgument(self.end_name, self, max)
+        return [size, start, end]
+
+    def promote(self, value):
+        if not isinstance(value, tuple):
+            size, start, end = self.rtargs
+            value = (value, start.default_value, value)
+        else:
```

Comment:
can we just have ``elif``s ? are we handling all cases ?

Reply:
The condition is on whether `value` is a `tuple` or not so technically, yes, we are handling all the cases. What is your concern here?

Reply:
gotcha now